site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1690,10 +1690,9 @@ public function getLateDayCacheForUserGradeable(string $user_id, string $g_id):
             }
         }
 
-        $row = $cache[$user_id];
-
+        $row = null;
         // If cache doesn't exist, generate it and query again
-        if (empty($row)) {
+        if (!array_key_exists($user_id, $cache) || empty($cache[$user_id])) {
```

Comment:
There's no need to change this now that it has been merged, but just FYI `empty()` does an `isset()` check for you.  The issue is that by setting `$row`, `$cache[$user_id]` is evaluated and an index not found error is thrown.

You could do this more concisely by using `if (empty($cache[$user_id])) {` alone.